### PR TITLE
Added Redis scaler

### PR DESCRIPTION
**Pull request by patnaikshekhar**
Pull request for Redis Scaler. [Issue 240](https://github.com/kedacore/keda/issues/240)

### CODE REVIEW

File: pkg/scalers/redis_scaler.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	listName         string
+	address          string
+	password         string
+}
+
+// NewRedisScaler creates a new redisScaler
+func NewRedisScaler(resolvedEnv, metadata map[string]string) (Scaler, error) {
+	meta, err := parseRedisMetadata(metadata, resolvedEnv)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing azure queue metadata: %s", err)
```
Comment (ahmelsayed) [inline pkg/scalers/redis_scaler.go]: Update the error message for redis.

File: pkg/scalers/redis_scaler.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	"github.com/go-redis/redis"
+	v2beta1 "k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+const (
+	listLengthMetricName    = "RedisLengthOfQueue"
```
Comment (yaron2) [inline pkg/scalers/redis_scaler.go]: `RedisQueueLength` would be a better pick here..

File: pkg/scalers/redis_scaler.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	defaultTargetListLength = 5
+	defaultRedisAddress     = "redis"
```
Comment (yaron2) [inline pkg/scalers/redis_scaler.go]: Why is this a default redis address? Its not a network addressable value..?

File: pkg/scalers/redis_scaler.go (hunk 1 above)
Comment (yaron2) [inline pkg/scalers/redis_scaler.go]: It says here azure queue.. change to redis list.

Comment (patnaikshekhar) [reply pkg/scalers/redis_scaler.go]: @ahmelsayed thanks. I've updated the error message now.

File: pkg/scalers/redis_scaler.go (hunk 2 above)
Comment (patnaikshekhar) [reply pkg/scalers/redis_scaler.go]: @yaron2 thanks! I've updated the metric name to "RedisListLength" now

File: pkg/scalers/redis_scaler.go (hunk 3 above)
Comment (patnaikshekhar) [reply pkg/scalers/redis_scaler.go]: @yaron2 added a default address of redis-master.default.svc.cluster.local:6379

File: pkg/scalers/redis_scaler.go (hunk 1 above)
Comment (patnaikshekhar) [reply pkg/scalers/redis_scaler.go]: @yaron2 updated the error message.

**Pull request merged**